stream/src/main/scala/org/apache/pekko/stream/scaladsl/Flow.scala

```diff
@@ -1779,6 +1779,46 @@ trait FlowOps[+Out, +Mat] {
   def groupedWeighted(minWeight: Long)(costFn: Out => Long): Repr[immutable.Seq[Out]] =
     via(GroupedWeighted[Out](minWeight, costFn))
 
+  /**
+   * Partitions this stream into chunks by a delimiter function, which is applied to each incoming element,
+   * when the result of the function is not the same as the previous element's result, a chunk is emitted.
+   *
+   * The `f` function must return a non-null value for all elements, otherwise the stage will fail.
+   *
+   * '''Emits when''' the delimiter function returns a different value than the previous element's result
+   *
+   * '''Backpressures when''' a chunk has been assembled and downstream backpressures
+   *
+   * '''Completes when''' upstream completes
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * @since 1.2.0
+   */
+  def groupedAdjacentBy[T](f: Out => T): Repr[immutable.Seq[Out]] =
+    via(GroupedAdjacentByWeighted(f, Long.MaxValue, ConstantFun.oneLong))
+
+  /**
+   * Partitions this stream into chunks by a delimiter function, which is applied to each incoming element,
+   * when the result of the function is not the same as the previous element's result, or the accumulated weight exceeds
+   * the `maxWeight`, a chunk is emitted.
+   *
+   * The `f` function must return a non-null value , and the `costFn` must return a non-negative result for all inputs,
+   * otherwise the stage will fail.
+   *
+   * '''Emits when''' the delimiter function returns a different value than the previous element's result, or exceeds the `maxWeight`.
+   *
+   * '''Backpressures when''' a chunk has been assembled and downstream backpressures
+   *
+   * '''Completes when''' upstream completes
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * @since 1.2.0
+   */
+  def groupedAdjacentByWeighted[T](f: Out => T, maxWeight: Long)(costFn: Out => Long): Repr[immutable.Seq[Out]] =
```

Comment:
is there a reason not to support methods on the scaladsl Source, SubFlow and SubSource like we have for javadsl?

Reply:
scaladsl'SubFlow extends FlowOps, so it's visible to scaladsl's Source/SubFlow/SubSource.